docs/dev_guide.md

```diff
@@ -1,31 +1,166 @@
 # Dev Guide
 
-## Adding a new param in GCSFuse
+## Adding new features to GCSFuse
+
+**If you're interested in adding a new feature, please open a GitHub issue to
+discuss your proposal with the GCSFuse team first before making the
+contribution.**
```

Comment:
'Make sure to describe the use case (what are you trying to solve and why?), and the proposed solution.

Reply:
Also add: We make no guarantees that contribution requests are accepted, or will be rolled out as a Generally Available feature.